index.js

```diff
@@ -0,0 +1,56 @@
+
+function calcTotalSalary(lineup) {
+  const totalSalary = lineup.reduce((salary, player) => {
+    return salary + player.salary
+  }, 0)
+
+  return totalSalary <= 45000
+}
+
+function getTeamCount(lineup) {
+  const playersPerTeam = lineup.reduce((teams, player) => {
+    teams[player.teamId] = teams[player.teamId] === undefined ? 1 : teams[player.teamId] + 1
+
+    return teams
+  }, {})
+
+  return !Object.values(playersPerTeam).some((teamLimit) => { return teamLimit > 2 })
+}
+
+function getGameCount(lineup) {
+  const playersPerGame = lineup.reduce((games, player) => {
+    games[player.gameId] = games[player.gameId] === undefined ? 1 : games[player.gameId] + 1
+
+    return games
+  }, {})
+
+  return !Object.values(playersPerGame).some((gameLimit) => { return gameLimit > 3 })
+}
+
+function getPositionCount(lineup) {
+  const singlePositions = ['P', 'C', '1B', '2B', '3B', 'SS']
+  const allPositions = lineup.reduce((position, player) => {
+    position[player.position] = position[player.position] === undefined ? 1 : position[player.position] + 1
+
+    return position
+  }, {})
+
+  for (let position in allPositions) {
+    if (position === 'OF' && allPositions[position] !== 3) {
+      return false
+    } else if (singlePositions.includes(position) && allPositions[position] !== 1) { return false }
+  }
+
+  return true
+}
```

Comment:
This is a lot of extra looping. You've already looped over the array in the reduce. Now you're looping over the positions array and then inside that using includes which also loops over it. Could be refactored to use fewer loops.